src/ElementPortal.js

```diff
@@ -3,16 +3,16 @@ import React from 'react';
 import ReactDOM from 'react-dom';
 import createReactClass from 'create-react-class';
 
+const { slice } = Array.prototype;
+
 const noop = () => {};
 
 const ElementPortal = createReactClass({
   propTypes: {
-    id: PropTypes.string,
```

Comment:
I'm a little torn on getting rid of this prop, since it seems like such a typical usage, and it gives parity with `getElementById`, but I agree it's not strictly necessary. If we wanted to avoid the collision, we could call it `byId`. But simplifying the API and code is probably worth the extra noise in usage. Do you happen to know if `getElementById('foo')` and `querySelectorAll('#foo')` perform similarly? I'm _guessing_ the id case is pretty optimized, but I don't personally know.

Reply:
> I'm a little torn on getting rid of this prop, since it seems like such a typical usage, and it gives parity with getElementById, but I agree it's not strictly necessary.

I get your point. I'm not sure people use so much `id`s those days 🤔

> If we wanted to avoid the collision, we could call it byId. But simplifying the API and code is probably worth the extra noise in usage.

Yes I think every *utility* feature should sit in userland, less complexity our side.

> Do you happen to know if getElementById('foo') and querySelectorAll('#foo') perform similarly? I'm guessing the id case is pretty optimized, but I don't personally know.

I honestly don't know, but my guess would be that they are very similar, they must have a fast-path in `querySelectorAll` for ids. But even if `querySelectorAll` were slower, the fact that we merge arrays to collect all matches node must compensate the *little* speed boost of `getElementById`. Plus it removes code, which is great for us.

Reply:
Another idea (borrowed from [react-portal](https://github.com/tajo/react-portal#portal)), we could just have a `node` prop and let the user select it however he wants.

Reply:
@jdeal After some thought, I'm really into that `node` prop and let the user feed it however he wants. What do you think?

Reply:
@ngryman It's not just a single _node_ though. It would have to be `nodes` right? `react-element-portal` is meant to be able to be able to target _many_ nodes and use data attributes from them. Next up, you'll tell me it should target a single node, and you should do your own `.map` to target multiple. 😛

Not sure where we draw the line, but this library is meant to make it simple to mix client-side react with server-side markup. There's a point at which you push everything into user land, and it really doesn't do much of anything anymore.

Reply:
> It's not just a single node though. It would have to be nodes right? [...] Next up, you'll tell me it should target a single node, and you should do your own .map to target multiple. 😛

@jdeal Haha, why not 😆? More seriously, it depends on how much sugar you want to add. More sugar means more easy to use, but also more complex and sometimes less generic. What I like in the `node(s)` proposal is that:
 1. We keep it generic, so you can do whatever you want.
 2. We align on what `react-portal` does, which is a good thing for people potentially migrating.

However, I get your point, having to type `querySelectorAll` in the middle of your JSX makes things less idiomatic. So I'm proposing a compromise: supporting both?

While `selector` in a nice shorthand in a JSX context, `node` offers more freedom and could be adapted when using the higher-order version of this component.

What do you think?

Reply:
Yep, I'm fine with both options @ngryman.